styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java

```diff
@@ -46,29 +46,26 @@
   private static final Logger LOG = LoggerFactory.getLogger(DockerRunnerHandler.class);
 
   private final DockerRunner dockerRunner;
-  private final StateManager stateManager;
 
   public DockerRunnerHandler(
-      DockerRunner dockerRunner,
-      StateManager stateManager) {
+      DockerRunner dockerRunner) {
     this.dockerRunner = requireNonNull(dockerRunner);
-    this.stateManager = requireNonNull(stateManager);
   }
 
   private boolean isUserError(Throwable e) {
     return e instanceof InvalidExecutionException;
   }
 
   @Override
-  public void transitionInto(RunState state) {
+  public void transitionInto(RunState state, EventRouter eventRouter) {
     switch (state.state()) {
       case SUBMITTING:
         final RunSpec runSpec;
         try {
           runSpec = createRunSpec(state);
         } catch (ResourceNotFoundException e) {
           LOG.error("Unable to start docker procedure.", e);
-          stateManager.receiveIgnoreClosed(Event.halt(state.workflowInstance()), state.counter());
+          eventRouter.receiveIgnoreClosed(Event.halt(state.workflowInstance()), state.counter());
```

Comment:
Why not make eventRouter a class variable again, instead of passing it on every event transition and have all output handlers change method signature?

Reply:
This is what this todo (https://github.com/spotify/styx/pull/788/files#diff-a3fabd4f5e1e00cf48f40dfdfa3356f1L361) says to fix.

Reply:
ah I understand now, thanks 👍